libsolidity/ast/AST.h

```diff
@@ -851,6 +851,12 @@ class CallableDeclaration: public Declaration, public VariableScope
 	ASTPointer<OverrideSpecifier> const& overrides() const { return m_overrides; }
 	std::vector<ASTPointer<VariableDeclaration>> const& returnParameters() const { return m_returnParameters->parameters(); }
 	ParameterList const& parameterList() const { return *m_parameters; }
+	SourceLocation const prototypeLocation(){
+		SourceLocation source = location();
+		SourceLocation functionLocation = parameterList().location();
+		source.end = functionLocation.end;
+		return source;
+	}
```

Comment:
You added the function but did not actually make anything use it. Please make sure that your changes actually have effect on compiler's output :)

I guess it would be easier to realize if we actually had tests for this case (if your change works and you don't update tests, they should be failing). I'd suggest adding a new [command-line test](https://github.com/ethereum/solidity/tree/develop/test/cmdlineTests).

Reply:
Thank you for your Review. From next time onwards I will keep in mind the review checklist.  Could you please tell me how to write test or it would be better if you direct me to some resources to learn about testing.

Reply:
For future reference, here's my answer from the channel:

> There's not much docs on that, but this should be relatively straightforward if you look at how existing tests look like
> Basically, there are two types of such tests: for the CLI (with `input.sol` or `input.yul` in them) and for Standard JSON (with `input.json` in them).
> Each test has some sample input and expected output.
> A file called `args` specifies the command-line options to pass to the compiler.
> Optionally there might be also be an `err` file specifying the errors we expect at standard error output.
> I'd suggest to start by creating a simple CLI test.
> Create `input.sol` that causes an overload error similar to what was shown in the issue.
args does not need to have anything in it.
> Then run `test/cmdlineTests.sh --update`.
> This will run all existing command line tests and your new one.
> The `--update` flag will make it automatically generate expected output if the one in the test is missing or does not match the actual output.
> Then you inspect that and, if it's fine, add it to the PR.

Reply:
> You added the function but did not actually make anything use it. Please make sure that your changes actually have effect on compiler's output :)

Note that this part of my comment has not been addressed yet :)